ruspy/simulation/simulation_auxiliary.py

```diff
@@ -213,22 +50,56 @@ def simulate_strategy_loop_known(
         :num_states: (int) : The size of the state space.
 
     """
+    num_states = ev.shape[0]
+    num_periods = decisions.shape[1]
     for period in range(num_periods):
-        for bus in range(num_buses):
-            old_state = states[bus, period]
-            if (-costs[old_state, 0] + unobs[bus, period, 0] + beta * ev[old_state]) > (
-                -costs[0, 0] - costs[0, 1] + unobs[bus, period, 1] + beta * ev[0]
-            ):
-                decision = 0
-                utility = -costs[old_state, 0] + unobs[bus, period, 0]
-                new_state = old_state + increments[bus, period]
-            else:
-                decision = 1
-                utility = -costs[0, 0] - costs[0, 1] + unobs[bus, period, 1]
-                new_state = increments[bus, period]
-
-            decisions[bus, period] = decision
-            utilities[bus, period] = utility
-            if period < num_periods - 1:
-                states[bus, period + 1] = new_state
+        old_state = states[bus, period]
+        if (-costs[old_state, 0] + unobs[bus, period, 0] + beta * ev[old_state]) > (
```

Comment:
Please create temporary variables here first that have good names such as `value_replace` and `value_maintenance` for example.